src/bt_peer_connection.cpp

```diff
@@ -2166,10 +2166,11 @@ namespace libtorrent
 		if (should_log(peer_log_alert::outgoing_message))
 		{
 			std::string bitfield_string;
-			bitfield_string.resize(num_pieces);
-			for (int k = 0; k < num_pieces; ++k)
+			std::size_t const n_pieces = aux::numeric_cast<std::size_t>(num_pieces);
+			bitfield_string.resize(n_pieces);
+			for (std::size_t k = 0; k < n_pieces; ++k)
```

Comment:
loops with unsigned counters variables make me nervous.. this is for indexing into the string?